tests/YearTest.php

```diff
@@ -35,6 +35,24 @@ public function testOfInvalidYearThrowsException(int $invalidYear): void
         Year::of($invalidYear);
     }
 
+    /**
+     * @dataProvider providerParse
+     */
+    public function testParse(string $string, int $expectedYear): void
+    {
+        self::assertYearIs($expectedYear, Year::parse($string));
+    }
+
+    public function providerParse(): array
+    {
+        return [
+            ['-2023', -2023],
+            ['-0100', -100],
+            ['1987', 1987],
+            ['121241', 121241],
+        ];
+    }
+
```

Comment:
Can you please add a `testParseInvalidYearThrowsException()` with a couple data such as `''`, `'9999999999'`, `'-100'`, `'+2000'`, `'ABC'`?

Reply:
Done